### DISCUSSION TITLE
[Documentation – Self hosting] Provide the proper Docker context

### DISCUSSION THREAD

**Discussion by rowild** [directus/directus#19246] [Feature Requests]
### Summary

I would like to propose an idea to improve the documentation a bit, specifically the update section.

The whole self hosting chapter is based on the Docker distribution, which is the only one, that the Directus team supports. However, the documentation only mentions the npm way to update/upgrade directus. This is confusing, since the Docker instance actually requires to pull the latest instance from Docker. But then it gets even more confusing, because where do I perform the `directus migrate:latest` command?

### Basic Example

Simply going to the command line and entering

`npx directus migrate:latest`

will create conflicts and eventually to a not-working instance of the Directus project. It is inevitable to tell the user at some point that the Docker instance must be entered:

`docker exec -it <directus-instance-name> sh`

and perform the update/upgrade command there.

Or, if that is not true, mention that pulling the latest Docker instance automatically performs the update migration. (It does not in my experience...)

### Motivation

Improve the documentation; clarify how to use nom with Docker; clarify how exactly to update: pulling the latest Docker image or using npm (and if so, where)

### Detailed Design

There should be a section before the first npm command example that describes sth like this:

`If you want to update your self-hosted Directus project, be aware that eecuting npm/npx commands on your regular command line won't have any effect on your Docker instance. In order to tell your Directus instance to update, you must first enter the shell of your Docker instance. Do so by:

`docker exec -it <directus-instance-name> sh`

A new prompt will appear, which then accepts the npm/npx command listed below.`

Of course this still does not explain what to do with the Docker image... this is the point where I am very confused...

### Must Have:
- Clarification when to use what to update the Directus instance (Docker image or npm)
- Clarification when and where to run `migrate:latest`

### Drawbacks

I wouldn't know of any.

### Alternatives

I wouldn't not of an alternative in this case.

### Adoption Strategy

Before I can provide a PR with text that can be discussed, I first would need to know, how the update process for self hosted Docker projects REALLY work. If I am NOT the one who provides the initial text, I would definitely like to be part of the review process.

### Unresolved Questions

None. (But is there a better interface to propose something for the documentation? It seems this interface is rather complicated for doc-related issues...)

**Comment by DanielBiegler** [comment]
Hi there! The docs do mention how to update with Docker inside the [Docker Guide](url) page, specifically [here](https://docs.directus.io/self-hosted/docker-guide.html#updating-with-docker-compose) under "Updating With Docker Compose".

> The specified image will be pulled and the containers recreated. Migrations will happen automatically so once the containers have started you will be on the latest version (or the version you specified).

Migrations happen automatically, you dont need to run the npm commands yourself.

I'll be closing this for now since I feel like this is solved. We can reopen this if you have something more specific in mind. Cheers! :)

**Reply by rowild** [reply]
Thank you! Excellent! – May I suggest then to put a little hint in the upgrade chapter, sth like `For upgrading a Docker instance, read up here: <link>`? 
Since everything is in separate chapters (Quickstart, Config Options, Docker Guide...) I was convinced that the Upgrade procedure is discussed in a separate chapter as well. So I didn't even think about going through the Docker section. IMO that little link would help... What do you think?

**Reply by rowild** [reply]
the "Updating..." section does not even have a separate submenu link...
